File: OffByOne/src/OffByOne.Qls/Ast/Style/Literals/StringLiteral.cs
namespace OffByOne.Qls.Ast.Style.Literals
{
    using System;

    using OffByOne.Qls.Ast.Style.Literals.Base;
    using OffByOne.Qls.Common.Visitors.Contracts;

    public class StringLiteral : Literal
    {
        public StringLiteral(string value)
        {
            if (value == null)
            {
                throw new ArgumentNullException(nameof(value), "A non-null string must be given");
            }

            this.Value = value.Trim('"');
        }

        public string Value { get; private set; }

        public override TResult Accept<TResult, TContext>(
            ILiteralVisitor<TResult, TContext> visitor,
            TContext context)
        {
            return visitor.Visit(this, context);
        }
    }
}

File: vanVlaanderenDerksen/QLJava/src/sc/ql/model/expressions/literals/StringLiteral.java
package sc.ql.model.expressions.literals;

import sc.ql.model.Expression;
import sc.ql.model.visitors.ExpressionVisitor;

public class StringLiteral extends Expression {
	private final String value;
	
	public StringLiteral(String value) {
		this.value = value;
	}
	
	public String getValue() {
		return this.value;
	}

	@Override
	public <T> T accept(ExpressionVisitor<T> visitor) {
		return visitor.visit(this);
	}
}
